src/connection.js

```diff
@@ -860,7 +916,18 @@ class Connection extends EventEmitter {
     this.debug.payload(function() {
       return preloginPayload.toString('  ');
     });
-
+    if (this.fedAuthInfo.method != undefined) {
+      if (0 != preloginPayload.fedAuthRequired && 1 != preloginPayload.fedAuthRequired) {
+        this.emit('connect', ConnectionError('Server sent an unexpected value for FedAuthRequired PreLogin Option. Value was' + preloginPayload.fedAuthRequired, 'EFEDAUTH'));
+        return this.close();
+      }
+      // fedAuthRequired is used for capability negotiation when choosing between SSPI and federated authentication
+      if (preloginPayload.fedAuthRequired === 1 && this.config.domain) {
```

Comment:
What do you think about adding TODO/Note here so that we remember to extend this condition to include other SSPI auths (Windows integrated and Kerberos) in future.